main_with_gui.py

```diff
@@ -172,23 +189,32 @@ def generate_files(self, window: QMainWindow, has_dedicated_gpu: bool):
 
 	def load_previously_generated_files(self, window:QMainWindow):
 		# if this is called the files surely exist (if not, the button was disabled)
-		with open(os.path.join(os.getcwd(), "tmp", "has_dedicated_gpu.txt")) as f:
-			has_dedicated_gpu = bool(f.read())
+		with open(os.path.join(os.getcwd(), "tmp", "gpu_loc.txt")) as f:
```

Comment:
I think the output file is called `gpu_location.txt`. To simplify the reference of a string in multiple points of your code, please add a constant at the top (`gpu_loc_file = "gpu_location.txt"`) and use it in the whole script.
```suggestion
		with open(os.path.join(os.getcwd(), "tmp", "gpu_location.txt")) as f:
```